cg/models/demultiplex/flow_cell.py

```diff
@@ -77,7 +77,26 @@ def full_name(self) -> str:
     @property
     def sample_sheet_path(self) -> Path:
         """Return sample sheet path."""
-        return Path(self.path, DemultiplexingDirsAndFiles.SAMPLE_SHEET_FILE_NAME)
+        root_sample_sheet_path: Path = Path(
+            self.path, DemultiplexingDirsAndFiles.SAMPLE_SHEET_FILE_NAME
+        )
+
+        unaligned_sample_sheet_path: Path = Path(
+            self.path,
+            DemultiplexingDirsAndFiles.UNALIGNED_DIR_NAME,
+            DemultiplexingDirsAndFiles.SAMPLE_SHEET_FILE_NAME,
+        )
+
+        sample_sheet_paths: List[Path] = [
+            root_sample_sheet_path,
+            unaligned_sample_sheet_path,
+        ]
+
+        for sample_sheet in sample_sheet_paths:
+            if sample_sheet.exists():
+                return sample_sheet
+
+        return root_sample_sheet_path
```

Comment:
```suggestion
   get_file_in_directory(Path = directory, file_name= "SampleSheet.csv")
```

Would return any sample sheet in the directory or sub directories, no need for loops or conditions.

Reply:
Yeah, that is nice! This property is used in some contexts where the sample sheet does not exist though. But we could use this and then return a default path if it does not exist. Going to think about it

Reply:
I'm keeping it as is for now, we can refactor it in a separate PR if we think it is worth it. I prefer not searching through the directory when we that it is present in either of two possible spots. We will be able to change simplify it once we have fixed https://github.com/Clinical-Genomics/cg/issues/2246 in the new flow.